Expensify/App
# [CP Staging] Revert "fix: treat certain US numbers as invalid"

### Details
Reverts Expensify/App#31726 and https://github.com/Expensify/App/pull/31010

We have been trying to trace down the rootcause of App crashing for some Expensifiers when opening a search component and these two PRs seem to be the root cause. 

When the search page is opened, the option rows are rendered and we run the checks for phone numbers in case the user has a primary login as phone number. Adding this regex and check in to the loop seems to have increased the load so much that for the biggest of account who know many users with phone number as their primary login this



### Fixed Issues

$ https://github.com/Expensify/App/issues/33326
PROPOSAL: N/A


### Tests

1. Open the search page
2. Verify it does not crash 

### Offline tests

N/A

### QA Steps

Will be tested by the largest accounts in Expensify to make sure the Search page is not crashing the App

### PR Author Checklist


### Screenshots/Videos

Straight reverts, tested in the Slack thread linked above

<details>
<summary>Android: Native</summary>






</details>

<details>
<summary>Android: mWeb Chrome</summary>






</details>

<details>
<summary>iOS: Native</summary>






</details>

<details>
<summary>iOS: mWeb Safari</summary>






</details>

<details>
<summary>MacOS: Chrome / Safari</summary>




</details>

<details>
<summary>MacOS: Desktop</summary>






</details>
